h5netcdf/core.py

```diff
@@ -622,12 +656,25 @@ def __init__(self, path, mode='a', invalid_netcdf=None, **kwargs):
         # unlimited), current size (identical to size for limited dimensions),
         # their position, and look-up for HDF5 datasets corresponding to a
         # dimension.
+        if phony_dims is not None:
+            self._phony_dim_count = 0
+        else:
+            self._phony_dim_count = None
         self._dim_sizes = ChainMap()
         self._current_dim_sizes = ChainMap()
         self._dim_order = ChainMap()
         self._all_h5groups = ChainMap(self._h5group)
-
         super(File, self).__init__(self, self._h5path)
+        # initialize all groups to detect/create phony dimensions
+        # mimics netcdf-c style naming
+        if phony_dims == 'sort':
+            self._determine_phony_dimensions()
```

Comment:
This should check to make use `phony_dims == 'access'` otherwise (or raise an error if not). Otherwise typos don't get checked.

Reply:
Sure, will add the check.